### ISSUE TITLE
Persistent naming rules are not generated for ENA devices

### ISSUE THREAD

**Issue by sam-aws** [amazonlinux/amazon-ec2-net-utils#48]
It looks like the latest `ec2-net-utils` in Amazon Linux 2 does not correctly generate `/etc/udev/rules.d/70-persistent-net.rules` when new interfaces are added.

## Bad case
On an `m5.large` running the latest AL2 AMI, the persistent naming rules don't run in response to attaching two interfaces:
```
[ec2-user@ip-172-31-4-222 ~]$ ip link  
​1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN mode DEFAULT group default qlen 1000  
   link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00  
​2: eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 9001 qdisc mq state UP mode DEFAULT group default qlen 1000  
   link/ether 0a:49:16:62:c5:1f brd ff:ff:ff:ff:ff:ff  
​4: eth1: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 9001 qdisc mq state UP mode DEFAULT group default qlen 1000  
   link/ether 0a:11:9f:00:f4:8b brd ff:ff:ff:ff:ff:ff  
​5: eth2: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 9001 qdisc mq state UP mode DEFAULT group default qlen 1000  
   link/ether 0a:94:de:7d:70:79 brd ff:ff:ff:ff:ff:ff
```
```
[ec2-user@ip-172-31-4-222 ~]$ ll /etc/udev/rules.d/  
total 32  
-rw-r--r-- 1 root root  294 Feb 14 20:22 51-ec2-hvm-devices.rules  
-rw-r--r-- 1 root root  575 Feb 14 20:22 51-ec2-xen-vbd-devices.rules  
-rw-r--r-x 1 root root  300 Feb 14 20:22 52-ec2-vcpu.rules  
-rw-r--r-- 1 root root  442 Mar 18 21:00 53-ec2-network-interfaces.rules  
-rw-r--r-- 1 root root  487 Feb 14 20:22 53-ec2-read-ahead-kb.rules  
-rw-r--r-- 1 root root  337 Feb 14 20:22 60-cdrom_id.rules  
-rw-r--r-- 1 root root 1603 Feb 14 20:22 70-ec2-nvme-devices.rules  
-rw-r--r-- 1 root root 1218 Mar 18 21:00 75-persistent-net-generator.rules
```

I suspect this is due to the change introduced in https://github.com/amazonlinux/amazon-ec2-net-utils/pull/21, i.e.
```

# do not create rule if interface uses unrecognized driver
ENV{ID_NET_DRIVER}!="vif|ena|ixgbevf", GOTO="persistent_net_generator_end"
```

These devices use the `ena` driver, but the udev event does not include that information:
```
[ec2-user@ip-172-31-4-222 ~]$ lspci -k | grep -A 2 Ethernet  
​00:05.0 Ethernet controller: Amazon.com, Inc. Elastic Network Adapter (ENA)  
       Kernel driver in use: ena  
       Kernel modules: ena  
​00:06.0 Ethernet controller: Amazon.com, Inc. Elastic Network Adapter (ENA)  
       Kernel driver in use: ena  
       Kernel modules: ena  
​00:07.0 Ethernet controller: Amazon.com, Inc. Elastic Network Adapter (ENA)  
       Kernel driver in use: ena  
       Kernel modules: ena  
```
```
KERNEL[368.302913] add      /devices/pci0000:00/0000:00:06.0/net/eth1 (net)  
ACTION=add  
DEVPATH=/devices/pci0000:00/0000:00:06.0/net/eth1  
IFINDEX=4  
INTERFACE=eth1  
SEQNUM=1432  
SUBSYSTEM=net
```

## Good case
If I downgrade `ec2-net-utils` to `1.5-3.amzn2`, the rules are generated:
```
[ec2-user@ip-172-31-12-135 ~]$ ip link  
​1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN mode DEFAULT group default qlen 1000  
   link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00  
​2: eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 9001 qdisc mq state UP mode DEFAULT group default qlen 1000  
   link/ether 0a:c6:95:75:f4:81 brd ff:ff:ff:ff:ff:ff  
​3: eth1: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 9001 qdisc mq state UP mode DEFAULT group default qlen 1000  
   link/ether 0a:11:9f:00:f4:8b brd ff:ff:ff:ff:ff:ff 
```
```
[ec2-user@ip-172-31-12-135 ~]$ ls /etc/udev/rules.d/  
51-ec2-hvm-devices.rules      52-ec2-vcpu.rules                53-ec2-read-ahead-kb.rules  70-ec2-nvme-devices.rules  75-persistent-net-generator.rules  
51-ec2-xen-vbd-devices.rules  53-ec2-network-interfaces.rules  60-cdrom_id.rules           70-persistent-net.rules  
```
```
[ec2-user@ip-172-31-12-135 ~]$ cat /etc/udev/rules.d/70-persistent-net.rules    
# This file was automatically generated by the /usr/lib/udev/write_net_rules  
# program, run by the persistent-net-generator.rules rules file.  
#  
# You can modify it, as long as you keep each rule on a single  
# line, and change only the value of the NAME= key.  
  
# elastic network interface  
SUBSYSTEM=="net", ACTION=="add", DRIVERS=="?*", ATTR{address}=="0a:11:9f:00:f4:8b", KERNEL=="eth*", NAME="eth1"  
  
# elastic network interface  
SUBSYSTEM=="net", ACTION=="add", DRIVERS=="?*", ATTR{address}=="0a:94:de:7d:70:79", KERNEL=="eth*", NAME="eth2"
```

**Comment by nmeyerhans** [comment]
I think the intent of the previous PR (#21) is best captured given the metadata associated with the initial hotplug event with
`DEVPATH=="/devices/virtual/*", GOTO="persistent_net_generator_end"`, as it matches things like bridge devices, vlans, etc.  Xen VIFs and PCI passthrough devices won't match the `/devices/virtual/` path and will not follow the GOTO.

**Comment by nmeyerhans** [comment]
I've confirmed that the `DEVPATH=="/devices/virtual/*"` test skips vlan devices (the specific topic that #21). Other interface types (e.g. bridge) are already skipped due to the existing `KERNEL!="eth*"` test. So my inclination is to go with that change to fix `75-persistent-net-generator.rules`.

I noticed that vlan interfaces trigger other issues related to the udev rules. In particular, the rules from `53-ec2-network-interfaces.rules` still fire for vlan devices, leading to systemd invoking `ec2ifup` for the device. However, since most of the work that `ec2ifup` does is driven by data from IMDS, the resulting configuration is incomplete. Unfortunately, it gets far enough into the interface configuration that it starts `dhclient`, which makes me suspect that fixing this problem may break things for some people.
